File: spring/core/SpringBeanLifecycles/src/main/java/demos/lifecycle/processors/PaymentEngineMock.java
package demos.lifecycle.processors;

import demos.lifecycle.common.PaymentEngine;
import org.springframework.stereotype.Component;

import javax.annotation.PostConstruct;
import javax.annotation.PreDestroy;

@Component("payment")
public class PaymentEngineMock implements PaymentEngine {
    public boolean authorize(String cardNo, double amount) {
        return amount < 1000;
    }

    @PostConstruct
    public void begin() {
        System.out.println("Spring context just created a payment engine");
    }

    @PreDestroy
    public void end() {
        System.out.println("Spring context just released a payment engine");
    }
}

File: spring/core/AnnotationDrivenDIWithKotlin/src/main/kotlin/demos/spring/notes/v9/PaymentEngineMock.kt
package demos.spring.notes.v9

import demos.spring.notes.common.PaymentEngine
import org.springframework.stereotype.Component

@Component("payment")
class PaymentEngineMock : PaymentEngine {
    override fun authorize(cardNo: String, amount: Double): Boolean {
        return amount < 1000
    }
}
